Sources/Moya/MoyaProvider.swift

```diff
@@ -150,11 +150,12 @@ public func convertResponseToResult(_ response: HTTPURLResponse?, request: URLRe
         case let (.some(response), data, .none):
             let response = Moya.Response(statusCode: response.statusCode, data: data ?? Data(), request: request, response: response)
             return .success(response)
-        case let (_, _, .some(error)):
-            let error = MoyaError.underlying(error)
+        case let (.some(response), _, .some(error)):
+            let response = Moya.Response(statusCode: response.statusCode, data: data ?? Data(), request: request, response: response)
```

Comment:
honestly, i'm not sure... But i can update to this to prevent any possible error... Let me know what you think

```swift
case let (.some(response), _, .some(error)):
            let response = Moya.Response(statusCode: response.statusCode, data: data ?? Data(), request: request, response: response)
            let error = MoyaError.underlying(error, response)
            return .failure(error)
case let (_, _, .some(error)):
            let error = MoyaError.underlying(error, nil)
            return .failure(error)
```